source/postprocess/visualization/theta_tauisa.cc

```diff
@@ -0,0 +1,215 @@
+
+        const QMidpoint<dim> quadrature_formula;
+        const unsigned int n_q_points = quadrature_formula.size();
+
+        FEValues<dim> fe_values(this->get_mapping(), this->get_fe(),
+                                quadrature_formula,
+                                update_values | update_gradients | update_quadrature_points);
+
+        // Set up material models
+        MaterialModel::MaterialModelInputs<dim> in(n_q_points,
+                                                   this->n_compositional_fields());
+        MaterialModel::MaterialModelOutputs<dim> out(n_q_points,
+                                                     this->n_compositional_fields());
+
+        // Set up cell iterator for looping
+        typename DoFHandler<dim>::active_cell_iterator cell =
+          this->get_dof_handler().begin_active(), endc =
+            this->get_dof_handler().end();
+
+        // Loop over cells and calculate tauISA in each one
+        // Note that we start after timestep 0 because we need the strain rate,
+        // which doesn't exist during the initial step
+        unsigned int cell_index = 0;
+        for (; cell != endc; ++cell, ++cell_index)
+          {
+            if (cell->is_locally_owned() && this->get_timestep_number() > 0)
+              {
+
+                // Fill the material model objects for the cell (for strain rate)
+                fe_values.reinit(cell);
+                in.reinit(fe_values, cell, this->introspection(),
+                          this->get_solution(), true);
+
+                // Calculate eigenvalues of strain rate and take maximum (absolute value)
+                // to get tauISA, the timescale for grain rotation toward the infinite strain axis
+                const SymmetricTensor<2, dim> strain_rate = in.strain_rate[0];
+                std::array<double, dim> strain_rate_eigenvalues = eigenvalues(
+                                                                    strain_rate);
+                double lamda1 = std::max(std::abs(strain_rate_eigenvalues[0]),
+                                         std::abs(strain_rate_eigenvalues[dim]));
+                double tauISA = 1.0 / lamda1;
+
+                (*return_value.second)(cell_index) = tauISA;
+              }
+          }
+
+        return return_value;
+      }
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(Theta, "Theta",
+                                                  "A visualization output object that generates output"
+                                                  "showing the angle between the infinite strain axis "
+                                                  "(approximated by a truncated series solution) "
+                                                  "and the flow velocity.")
```

Comment:
If you had one available, a formulaic description of these quantities would be much appreciated here.

The text you have here will eventually become input to latex, so you can use all latex comments for formulas, etc. The only thing to pay attention to is that in latex you use one `\`, whereas here you need to double up on all backslashes.

The same applied below for the other postprocessor.

Reply:
Also, much of the commentary you have in the big comment at the top of the .h file would fit very well here. Most people read the manual (which is generated using information such as the one here), not the doxygen generated documentation, so having the documentation here (or ideally in both places) is useful!